File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> targetMap = new HashMap<>();
        for (int i=0; i< nums.length; i++) {
            if (targetMap.containsKey(target-nums[i])) {
                return new int[] {i, targetMap.get(target-nums[i])};
            }
            targetMap.put(nums[i], i);
        }
        return new int[]{};
    }
}

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        numSet = set()
        for i, n in enumerate(nums):
            if target - n in numSet:
                return [nums.index(target - n), i]
            numSet.add(n)
        